### PR TITLE
Fill out `layoutSystemForChildren` in `fixSpyOnlyMetadata`

### PR DESCRIPTION
**Pull request by bkrmendy**
## Problem
Pasting an element into an empty conditional inside a flex container pastes the element as absolute in an awkward place

## Cause & fix
`layoutSystemForChildren` is not fixed up for conditionals, so the pasting logic thinks that it's a flow container with no global frame, so it bails out with a fallback and places the pasted element to top: 0, left: 0.

This is fixed by actually filling in `layoutSystemForChildren` for conditionals based on their ancestors (this is what `fillLayoutSystemForChildrenFromAncestors` does). This PR also changes the type of `SpecialSizeMeasurements.layoutSystemForChildren` to `DetectedLayoutSystem | null`.

Before vs after: https://screenshot.click/16-57-e1kpo-gh6fw.mp4

### CODE REVIEW

File: editor/src/components/canvas/__snapshots__/ui-jsx-canvas-bugs.spec.tsx.snap [hunk 1]
```diff
@@ -185,7 +185,7 @@ Object {
       },
       "immediateParentProvidesLayout": true,
       "justifyContent": null,
-      "layoutSystemForChildren": "flow",
+      "layoutSystemForChildren": null,
```
Comment (balazsbajorics) [inline editor/src/components/canvas/__snapshots__/ui-jsx-canvas-bugs.spec.tsx.snap:188]: 👌👌👌👌

File: editor/src/components/editor/actions/actions.spec.browser2.tsx [hunk 2]
```diff
@@ -3015,6 +3015,86 @@ export var storyboard = (props) => {
...
+          firePasteEvent(canvasRoot)
+
+          // Wait for the next frame
+          await clipboardMock.pasteDone
+          await renderResult.getDispatchFollowUpActionsFinished()
+
+          await pressKey('Esc')
+          await renderResult.getDispatchFollowUpActionsFinished()
+
+          // no position: absolute, top, left in the pasted image
```
Comment (balazsbajorics) [inline editor/src/components/editor/actions/actions.spec.browser2.tsx:3066]: you could expect() this from computedStyle or allElementProps!

**Review by balazsbajorics — APPROVED** [review]

File: editor/src/components/editor/actions/actions.spec.browser2.tsx (hunk 2 above)
Comment (bkrmendy) [reply editor/src/components/editor/actions/actions.spec.browser2.tsx]: great idea, https://github.com/concrete-utopia/utopia/pull/4374/commits/585aa88bb955ae5813611e7ac79607ef9cfe5514

**Review by gbalint — APPROVED** [review]

**Review by Rheeseyb — COMMENTED** [review]
Nice!

**Pull request merged**